src/components/Form.js

```diff
@@ -0,0 +1,182 @@
+      role="form"
+      ref={ref}
+    >
+      <InputField
+        label="name:"
+        role="input"
+        {...register('name', {
+          required: true,
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      {errors.name && 'Name is required'}
+      <InputField
+        label="phone:"
+        role="input"
+        {...register('phone', {
+          required: true,
+          min: 11,
+          max: 18,
+          patter: /^[0-9]+$/,
+        })}
+      />
+      {errors.phone && 'Phone is required'}
+      <InputField
+        label="email:"
+        role="input"
+        {...register('email', {
+          required: true,
+          pattern: /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/,
+        })}
+      />
+      {errors.email && 'Email is required'}
+      <InputField
+        label="department:"
+        role="input"
+        {...register('department', {
+          required: true,
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      {errors.department && 'Department is required'}
+      <InputField
+        label="skill1:"
+        role="input"
+        {...register('skill1', {
+          required: true,
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      {errors.skill1 && 'At least one skill is required'}
+      <InputField
+        label="skill:"
+        role="input"
+        {...register('skill2', {
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      <InputField
+        label="skill:"
+        role="input"
+        {...register('skill3', {
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      <InputField
+        label="skill:"
+        role="input"
+        {...register('skill4', {
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      <InputField
+        label="skill:"
+        role="input"
+        {...register('skill5', {
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      <InputField
+        label="skill:"
+        role="input"
+        {...register('skill6', {
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      <Button disabled={isDisabled} type="submit" children="Create User" />
+    </StyledForm>
+  )
+
+  function handleSubmit(event) {
+    event.preventDefault()
+    const form = event.target
+    const inputs = form.elements
+    const name = inputs.name.value
+    const phone = inputs.phone.value
+    const email = inputs.email.value
+    const department = inputs.department.value
+    const skill1 = inputs.skill1.value
+    const skill2 = inputs.skill2.value
+    const skill3 = inputs.skill3.value
+    const skill4 = inputs.skill4.value
+    const skill5 = inputs.skill5.value
+    const skill6 = inputs.skill6.value
+    const skills = [skill1, skill2, skill3, skill4, skill5, skill6]
```

Comment:
Are there fixed skills? So a user who has 7 skills can't add more? How about only having one input field for skills and the user can add multiple skills separated by comma? I think that would help to shorten this form and the complexity you have here.